src/browserbox-imap.js

```diff
@@ -111,12 +117,7 @@
      */
     ImapClient.prototype.connect = function() {
         return new Promise((resolve, reject) => {
-            this.socket = this._TCPSocket.open(this.host, this.port, {
-                binaryType: 'arraybuffer',
-                useSecureTransport: this.secureMode,
-                ca: this.options.ca,
-                tlsWorkerPath: this.options.tlsWorkerPath
-            });
+            this.socket = this._TCPSocket.open(this.host, this.port, this.options.tcpSocket);
 
```

Comment:
true that. then we should pass in all the [standardized socket options](https://developer.mozilla.org/en-US/docs/Web/API/TCPSocket) via the first options object, and the non-standard options for the shim in a second object (ca, ws, worker paths, ...). this would also require a change in the tcp-socket API, but that's a clean way to handle it.